bpmn-model/src/test/java/io/camunda/zeebe/model/bpmn/validation/ZeebeJobWorkerTaskValidationTest.java

```diff
@@ -16,153 +16,126 @@
+  void validStaticJobTypeAndRetries(final JobWorkerTaskBuilder taskBuilder) {
+
+    final BpmnModelInstance process =
+        processWithTask(taskBuilder, task -> task.zeebeJobType("service").zeebeJobRetries("5"));
+
+    ProcessValidationUtil.assertThatProcessIsValid(process);
+  }
+
+  @ParameterizedTest
+  @MethodSource("jobWorkerTaskBuilderProvider")
+  @DisplayName("task with job type and retries expression")
+  void validJobTypeAndRetriesExpression(final JobWorkerTaskBuilder taskBuilder) {
+
+    final BpmnModelInstance process =
+        processWithTask(
+            taskBuilder,
+            task ->
+                task.zeebeJobTypeExpression("serviceType").zeebeJobRetriesExpression("jobRetries"));
+
+    ProcessValidationUtil.assertThatProcessIsValid(process);
+  }
+
+  @ParameterizedTest
+  @MethodSource("jobWorkerTaskBuilderProvider")
+  @DisplayName("task with custom header")
+  void validCustomHeader(final JobWorkerTaskBuilder taskBuilder) {
+
+    final BpmnModelInstance process =
+        processWithTask(
+            taskBuilder, task -> task.zeebeJobType("service").zeebeTaskHeader("priority", "high"));
+
+    ProcessValidationUtil.assertThatProcessIsValid(process);
+  }
+
+  @ParameterizedTest
+  @MethodSource("jobWorkerTaskBuilderProvider")
+  @DisplayName("task without job type")
+  void missingJobType(final JobWorkerTaskBuilder taskBuilder) {
+
+    final BpmnModelInstance process = processWithTask(taskBuilder, task -> task.id("task"));
+
+    ProcessValidationUtil.assertThatProcessHasViolations(
+        process, expect("task", "Must have exactly one 'zeebe:taskDefinition' extension element"));
+  }
+
+  @ParameterizedTest
+  @MethodSource("jobWorkerTaskBuilderProvider")
+  @DisplayName("task with empty job type")
+  void emptyJobType(final JobWorkerTaskBuilder taskBuilder) {
+
+    final BpmnModelInstance process = processWithTask(taskBuilder, task -> task.zeebeJobType(""));
+
+    ProcessValidationUtil.assertThatProcessHasViolations(
+        process, expect(ZeebeTaskDefinition.class, "Task type must be present and not empty"));
+  }
+
+  private BpmnModelInstance processWithTask(
+      final JobWorkerTaskBuilder taskBuilder,
+      final Consumer<AbstractJobWorkerTaskBuilder<?, ?>> taskModifier) {
+
+    final StartEventBuilder processBuilder = Bpmn.createExecutableProcess("process").startEvent();
+    final AbstractJobWorkerTaskBuilder<?, ?> jobWorkerTaskBuilder =
+        taskBuilder.build(processBuilder);
+    taskModifier.accept(jobWorkerTaskBuilder);
+    return jobWorkerTaskBuilder.endEvent().done();
+  }
+
+  private static Stream<JobWorkerTaskBuilder> jobWorkerTaskBuilderProvider() {
+    return Stream.of(
+        JobWorkerTaskBuilder.of("serviceTask", AbstractFlowNodeBuilder::serviceTask),
+        JobWorkerTaskBuilder.of("businessRuleTask", AbstractFlowNodeBuilder::businessRuleTask),
+        JobWorkerTaskBuilder.of("scriptTask", AbstractFlowNodeBuilder::scriptTask));
+  }
```

Comment:
Can this use the more general: `io.camunda.zeebe.engine.util.JobWorkerTaskBuilderProvider`?

Reply:
No. Here, we're in the model API module. The other provider is in the engine module :sweat_smile:

Reply:
Aah, that makes sense as well. Sorry for the stupid question 😄